tower-http/src/compression/service.rs

```diff
@@ -82,16 +88,26 @@ impl<S> Compression<S> {
         self.accept.set_br(false);
         self
     }
+
+    /// Replaces the current compression filter, `DefaultCompressionFilter` being the default
+    pub fn with_compression_filter<CF: CompressionFilter>(self, compression_filter: CF) -> Compression<S, CF> {
```

Comment:
```suggestion
    /// Replace the current compression filter.
    ///
    /// The default predicate is [`DefaultCompressionFilter`] which disables compression of gRPC
    /// (gRPC has its own protocol specific compression system) and responses who's
    /// mime type starts with `image/`.
    ///
    /// # Example
    ///
    /// ```
    /// todo!();
    /// ```
    pub fn compression_filter<CF>(self, compression_filter: CF) -> Compression<S, CF> 
    where
        CF: CompressionFilter
    {
```

- Expands the docs a bit to explain what the default compression does. Do we always apply the standard filtering regardless if this has been customized by the user? For example I don't think we should ever compress gRPC responses.
- I don't believe we normally use `with_` prefix for builder methods like this. For example `Trace` has `make_span_with`, `on_request`, etc.
- I personally prefer `where` clauses to `Foo: SomeTrait` 🤷 
- Makes `DefaultCompressionFilter` a link.
- We should probably also have an example.